### ISSUE TITLE
Stuck UI with Long Files 

### ISSUE THREAD

**Issue by architeck** [JuergenFleiss/aTrain#41]
I'm transcribing 2+ hour long wav files and notice that the transcription is completed rather quickly on a Nvidia 3070 using large-v3-turbo model. I can see the completed transcript in the output folder, but the UI remains stuck on "transcribing".

The image below is shows less time, but luckily I checked the folders after my long file run for 5+ hour without and progress.

![image](https://github.com/user-attachments/assets/c25e6f1e-bf57-4d50-85b0-b8ed9fee4b01)

Edit: I'm actually noticing the stuck UI thing on smaller files as well now.

**Comment by roperna** [comment]
I've been facing the same issue in the last three days. I'm transcribing 1/1.5 hour long wav files. Compared to last week, I cannot select the model anymore (the only option now is 'large-v3-turbo'), and the UI remains stuck on "transcribing". 
I checked the folders after my files running for 10+ hours, but there are no progress.

**Comment by JuergenFleiss** [comment]
Ok, we released the new version 1.2 three days ago, so this is probably a bug that we did not catch in testing and I cannot reproduce it on my end right now. We will have to take a look on the new way we track the progress in the new steps. 

Large-v3 turbo is freshly released from OpenAI and offers great balance between speed and accuracy. As for the other models, you now have a new menu point on the left: Model Manager. There you can download all the available whisper models, also large-v2 or large-v3.

 Could you try other models and let me know if the UI is also stuck with those?

**Comment by SchwarzDeveloping** [comment]
Same here. It randomly gets stuck with files over 23 minutes long, while files under 20 minutes pose no issues. The file extension doesn't seem to matter.

**Comment by stevevaius2015** [comment]
Hello, Did you release v1.2 ? Where is the release to download for developers?

**Comment by JuergenFleiss** [comment]
> Hello, Did you release v1.2 ? Where is the release to download for developers?

Hi, we released the new 1.2 installer on microsoft store last week; I also did a release on github now.

**Comment by aereimer** [comment]
Hi @JuergenFleiss, thanks for the amazing software.
Just want to report that I got similar issues using both atrain 1.2 and atrain core 1.2 (using this command line --model "large-v3" --language "en" --speaker_detection --num_speakers "auto-detect" --device "GPU"  --compute_type "float16")
in my case, changing the model (turbo, large-v3) did not matter
It seems 1.1 is working fine

**Comment by JuergenFleiss** [comment]
Ok, thats quite strange.
Might be related to faster-whisper itself, they also have reports of videos >20min crashing
https://github.com/SYSTRAN/faster-whisper/issues/687

Could one of you share the files that crash atrain so that we can investigate (as we are unable to reproduce the problem)?

**Comment by happypig** [comment]
> Ok, thats quite strange. Might be related to faster-whisper itself, they also have reports of videos >20min crashing [SYSTRAN/faster-whisper#687](https://github.com/SYSTRAN/faster-whisper/issues/687)
> 
> Could one of you share the files that crash atrain so that we can investigate (as we are unable to reproduce the problem)? You can reach me under [bandas@uni-graz.at](mailto:bandas@uni-graz.at)

Hi @JuergenFleiss,

Thakns for your effort. Please check email for the file that crash atrain.

**Comment by JuergenFleiss** [comment]
Thanks for the file, @happypig I was able to reproduce the stuck UI issue with it, where it stops showing progress. The transcription completed fine in the background.  Could you check if you may in fact have the complete transcript in your documents folder even though it did show completed in the UI?

I am currently trying to disentangle whether there are any crashes, or it is just the UI issue.

@adlihm thanks for the info, i also watched during the transcription of @happypig  testfile and observed the same; once the actual transcription is completed ram usage drops but UI continues the timer. 

As for maybe having crashes with some segments (@aereimer), we will probably try to revert to faster-whisper backend of aTrain 1.1, where we also succesfully could run whisper v3 turbo in testing.

faster-whisper made a lot of changes in the recent releases that brought cool features (per segment language recognition e.g.) but we also got at least some few reports of decreased transcription quality in aTrain 1.2 compared to 1.1, especially missing text passages, which may be related to voice activation detection in newer faster-whisper versions (or something else), as they also have some issues open on this.

Did any of you experience decreased transcription quality?

**Comment by adlihm** [comment]
> @adlihm thanks for the info, i also watched during the transcription of @happypig testfile and observed the same; once the actual transcription is completed ram usage drops but UI continues the timer.

I just checked the transcription folder, and yep, the transcription is done, and the quality seems okay? I usually transcribed Indonesian, and honestly haven't checked every result in detailed. Just briefly, and it seems okay.

**Comment by happypig** [comment]
> Thanks for the file, @happypig I was able to reproduce the stuck UI issue with it, where it stops showing progress. The transcription completed fine in the background. Could you check if you may in fact have the complete transcript in your documents folder even though it did show completed in the UI?
> 
> I am currently trying to disentangle whether there are any crashes, or it is just the UI issue.
> 
> @adlihm thanks for the info, i also watched during the transcription of @happypig testfile and observed the same; once the actual transcription is completed ram usage drops but UI continues the timer.
> 
> As for maybe having crashes with some segments (@aereimer), we will probably try to revert to faster-whisper backend of aTrain 1.1, where we also succesfully could run whisper v3 turbo in testing.
> 
> faster-whisper made a lot of changes in the recent releases that brought cool features (per segment language recognition e.g.) but we also got at least some few reports of decreased transcription quality in aTrain 1.2 compared to 1.1, especially missing text passages, which may be related to voice activation detection in newer faster-whisper versions (or something else), as they also have some issues open on this.
> 
> Did any of you experience decreased transcription quality?

@JuergenFleiss thanks. 

The complete transcription was found in the folder even the UI stuck.

For the same 164 mins test file, the processing time for whisper v2, v3, and v3 turbo are 11 hrs, 10 hrs and 7 hrs. However, the quality are pretty amazing since v2 > v3 > v3 turbo. :p

**Comment by OhsawaSenpai** [comment]
I'm not sure if this is relevant, but in my scenario, the transcription files do not even show up, like the transcription failed. However, the logs state it is finished.

[metadata.txt](https://github.com/user-attachments/files/18019581/metadata.txt)
[log.txt](https://github.com/user-attachments/files/18019583/log.txt)

**Comment by JuergenFleiss** [comment]
> I'm not sure if this is relevant, but in my scenario, the transcription files do not even show up, like the transcription failed. However, the logs state it is finished. [metadata.txt](https://github.com/user-attachments/files/18019590/metadata.txt) [log.txt](https://github.com/user-attachments/files/18019591/log.txt)

yeah, that seems a different issue; thanks for adding the logs. Thats actually quite strange that they state completed. Is the problem reproducable or was it a one time thing?

@ArminHaberl is closing in on the stuck UI issue though.

**Comment by OhsawaSenpai** [comment]
I can confirm that this issue is reproducible with multiple files.

I’ve been unable to identify the root cause. Initially, I attempted to use a single 5-hour-long audio file, which exhibited the same behavior. To troubleshoot, I split the file into multiple 19-minute segments. While some of these files were successfully transcribed, about one-third consistently fail, and I’m unable to transcribe them at all.

All the files are similar in size. For the successful ones, the operation completes in 1–3 minutes. However, for the problematic files, the process gets stuck, though the logs always indicate the status as "completed."

@JuergenFleiss, would you like me to run additional tests or log a new issue for this?

**Comment by JuergenFleiss** [comment]
There where still some issues in 1.2.1; @ArminHaberl has now fixed those in the 1.2.2 branches of atrain and atrain_core

We will update Windows Store Version directly to 1.2.2 when no new issues arise in testing.

**Comment by JuergenFleiss** [comment]
Just an update. The newest branch 1.2.4 should be close to release ready and take care of the stuck UI issue and the problems with GPU transcription that have plagued us. 

You can test it as follows:

Setup a virtual environment
```
python -m venv venv
```
Activate the virtual environment
```
.\venv\Scripts\activate
```
Install aTrain
```
pip install git+https://github.com/JuergenFleiss/aTrain.git@v1.2.4 --extra-index-url https://download.pytorch.org/whl/cu121
```

Get required models:

```
aTrain init
```
Run the app with the console script
```
aTrain start
```

Just note that if you use **GPU with largev3-turbo currently you need to manually switch to "Float16" in the advanced settings**. We will change that in the next days to Float16 as default as for some reason v3-turbo freezes at the end of the transcription when used with Int8 (all other models work fine).

Thanks for all your patience. We will do the last changes outlined here in the next days and then test extensively before we hopefully release a smoothly working v1.2.4

**Comment by OhsawaSenpai** [comment]
Will be waiting for the v1.2.4 since I've been using the Windows Store version :)

Also, a question:
- Would it be possible to save the session setting values from the latest transcript? Example: every time I finish one file and want to do another, I've to manually change Language, Multi-speaker, and Computer Type manually.
- Are you thinking in implementing some kind of queuing mechanism in the future?

Thank you :)
